inst/tests/tests.Rraw

```diff
@@ -15194,6 +15194,78 @@ l = replicate(ceiling(log10(.Machine$integer.max)), rep(1L, 10L), simplify = FAL
+test(2062.015, fifelse(test_vec, list(1,3),list(2,4)), error = "Length of 'yes' and 'no' must be 1 or equal to length of 'test'.")
+test(2062.016, fifelse(test_vec, list(1), 0), as.list(out_vec))
+test(2062.017, fifelse(test_vec, 1, list(0)), as.list(out_vec))
+## Jan 1 - 5, 2011
+date_vec = as.Date(14975:14979, origin = '1970-01-01')
+test(2062.018,
+     fifelse(date_vec == "2011-01-01", date_vec - 1L, date_vec),
+     c(date_vec[1L] - 1L, date_vec[2:5]))
+test(2062.019,
+     fifelse(factor(letters[1:5]) %in% c("a","b","c"), factor(letters[1:5]), factor(NA)),
+     factor(c("a","b","c",NA,NA),levels = levels(factor(letters[1:5]))))
+test(2062.020, fifelse(test_vec_na, 1L, 0L), out_vec_na)
+test(2062.021, fifelse(test_vec_na, rep(1L,12L), 0L), out_vec_na)
+test(2062.022, fifelse(test_vec_na, rep(1L,12L), rep(0L,12L)), out_vec_na)
+test(2062.023, fifelse(test_vec_na, 1L, rep(0L,12L)), out_vec_na)
+test(2062.024, fifelse(test_vec_na, 1, 0), as.numeric(out_vec_na))
+test(2062.025, fifelse(test_vec_na, rep(1,12L), 0), as.numeric(out_vec_na))
+test(2062.026, fifelse(test_vec_na, rep(1,12L), rep(0,12L)), as.numeric(out_vec_na))
+test(2062.027, fifelse(test_vec_na, 1, rep(0,12L)), as.numeric(out_vec_na))
+test(2062.028, fifelse(test_vec_na, TRUE, rep(FALSE,12L)), as.logical(out_vec_na))
+test(2062.029, fifelse(test_vec_na, rep(TRUE,12L), FALSE), as.logical(out_vec_na))
+test(2062.030, fifelse(test_vec_na, rep(TRUE,12L), rep(FALSE,12L)), as.logical(out_vec_na))
+test(2062.031, fifelse(test_vec_na, "1", rep("0",12L)), as.character(out_vec_na))
+test(2062.032, fifelse(test_vec_na, rep("1",12L), "0"), as.character(out_vec_na))
+test(2062.033, fifelse(test_vec_na, rep("1",12L), rep("0",12L)), as.character(out_vec_na))
+test(2062.034, fifelse(test_vec_na, "1", "0"), as.character(out_vec_na))
+test(2062.035, fifelse(test_vec, as.Date("2011-01-01"), FALSE), error = "Item 'yes' is of type double but item 'no' is of type logical. Please make sure they are of the same type.")
+test(2062.036, fifelse(test_vec_na, 1+0i, 0+0i), as.complex(out_vec_na))
+test(2062.037, fifelse(test_vec_na, rep(1+0i,12L), 0+0i), as.complex(out_vec_na))
+test(2062.038, fifelse(test_vec_na, rep(1+0i,12L), rep(0+0i,12L)), as.complex(out_vec_na))
+test(2062.039, fifelse(test_vec_na, 1+0i, rep(0+0i,12L)), as.complex(out_vec_na))
+test(2062.040, fifelse(test_vec, as.raw(0), as.raw(1)), error = "Type raw is not supported.")
+test(2062.041, fifelse(TRUE,1,as.Date("2019-07-07")), error = "Item 'yes' has different class than item 'no'. Please make sure that they have same class type.")
+test(2062.042, fifelse(TRUE,1L,factor(letters[1])), error = "Item 'yes' has different class than item 'no'. Please make sure that they have same class type.")
+test(2062.043, fifelse(TRUE, list(1:5), list(5:1)), list(1:5))
+test(2062.044, fifelse(as.logical(NA), list(1:5), list(5:1)), list(as.integer(NA)))
```

Comment:
the `all(is.na(test))` case is interesting, can we add some more tests? I guess the behavior here is right, namely that the output should be `NA` & match the type of `yes`/`no`

Reply:
e.g. I notice:

```
dput(fifelse(NA, list(0), list(1)))
# list(NA_integer_)
```

Here `yes` is `list(numeric())` but the output is `list(integer())`.

I think this is fine (it would I think be overly complicated to match the recursive structure completely), but we should add a test & document.

Would `list(logical())` be preferred? Based on type heirarchy...